src/Univapay/Resources/TransactionToken.php

```diff
@@ -159,6 +159,12 @@ public function createCharge(
             : []
         );
 
+        if ($this->paymentType === PaymentType::CARD()) {
+            $payload += isset($threeDS)
+                ? ['three_ds' => $threeDS->jsonSerialize()]
```

Comment:
Instead of silently not adding the `three_ds` obj when it is not a card payment, we might want to additionally throw a `UnivapayLogicError` to warn the client that they are doing something wrong. We can still keep this keep this filter.

Additionally, add validation to ensure that

- if `cvv_authorize.enabled = true`, then `cvv_authorize.status == current`
- if `three_ds.enabled = true`, then `three_ds.status == successful`

Reply:
@kgston I understand the need for CVV validation, but it will introduce breaking changes to how the SDK is used. Currently, the CVV information is part of a different object when creating a charge or subscription. When calling `createCharge`, we will need the `token object` instead of just the `token_id`.

The `three_ds` situation is different because it requires customer intervention before we can capture the status. This process should be handled at the application level.

Reply:
The data should be in the `data` object though..

Reply:
my mistakes yes the object is there, add the validation